native_store.go

```diff
@@ -82,12 +94,18 @@ func (ns *nativeStore) Put(_ context.Context, serverAddress string, cred auth.Cr
 		dockerCred.Username = emptyUsername
 		dockerCred.Secret = cred.RefreshToken
 	}
-	return client.Store(ns.programFunc, dockerCred)
+	buffer := new(bytes.Buffer)
+	if err := json.NewEncoder(buffer).Encode(dockerCred); err != nil {
+		return err
+	}
+	_, err := ns.Execute(ctx, buffer, "store")
```

Comment:
Consider using `json.Marshal()` and `bytes.NewReader()`.

Reply:
Updated accordingly.